### ISSUE TITLE
Unlimited JCE Config?

### ISSUE THREAD

**Issue by nikb0yd** [codenvy/codenvy#1209]
I noticed that the alpine_jdk8/Dockerfile appears to have code for installation of the unlimited JCE. However, it's configured with JAVA_JCE=standard. Is there some easy way to externally override that and use JAVA_JCE=unlimited? Or, will I need to clone the code and build a new docker image with that change? Or, might there be one already available somewhere that I failed to find? Thanks

**Comment by TylerJewell** [comment]
Hi @nikb0yd - thanks for getting involved in the project!

So, it should be possible to override environment variables using Docker syntax. That particular image is stored in DockerHub as eclipse/alpine_jdk8.  

If you build a workspace using that image "FROM eclipse/alpine_jdk8" as the recipe, you can then use the stack editor in M7 when creating the workspace to define environment variables. You should be able to override the JAVA_JCE environment variable with a new value, which would set the value to your setting instead of 'standard'.

Now, I haven't personally tried this, as the stack editor is really new, but I know @ashumilova team has invested a lot of time on these scenarios, so that this can work just like I described.

**Comment by nikb0yd** [comment]
@TylerJewell thanks for the quick response. I was hoping something like that will be possible. I'd much rather re-use where possible. I'll try that out. Thanks again.

**Comment by nikb0yd** [comment]
@TylerJewell  Just a quick follow up. It took many failed experiments and a lot of reading (and reading between the lines), but I was finally able to create a stack derived from Alpine + JDK 8 with the requisite configuration (rather a steep learning curve). I look forward to some of the other work being done to allow more easy configuration of the environment for a stack. Predicated on my final success, I was able to get some of our custom library code built and tests run. Thanks again for the hints that pointed me in the right direction.

**Comment by TylerJewell** [comment]
@nikb0yd - Two things:
1) Hope you don't mind if we close this issue as it seems that your core question is resolved. However:
2) Would you be open to sharing all of your experiences and the modifications you had to make?  We have a lot of effort related to improving the stack editing, so the more we can collect your particular experiences the more we can build better specifications to minimize the sort of effort you have to put into these sorts of basic changes for the future.

Our goal here is that for 80% of the changes to an existing docker image or file, it should be really simple and easy to figure out.  and then the other 20%, we want you to write a full dockerfile within our IDE which will have dockerfile debugger.

**Comment by nikb0yd** [comment]
I cloned the Eclipse Che stack, and named it: Eclipse Che - Custom. Then, I edited the custom stack and added the following to the Raw Configuration after the "dev-machine" section:

      "recipe": {
          "contentType": "application/x-yaml",
          "type": "compose",
          "content": "services:\n dev-machine:\n  image: codenvy/alpine_jdk8\n  mem_limit: 2147483648\n  environment:\n   JAVA_JCE: unlimited\n"
      }

I then created a project using the custom stack, and examined the environment from within a shell after launching the stack. I then cloned one of our Maven projects from GitHub, and ran a build with tests to confirm that the crypto was working as expected.

Figuring out how to get the recipe added was one challenge. Finding an example of an existing recipe that used compose was another challenge. Once I saw those examples, I had to run a few experiments to confirm my understanding was working. The rest was fairly straightforward.
